### ISSUE TITLE
Authorizer AuthResponse routes with queryparameters not working in local

### ISSUE THREAD

**Issue by ivandjuricic** [aws/chalice#1470]
Sample app:
```
from chalice import Chalice, AuthResponse

app = Chalice(app_name='authpath')

@app.authorizer()
def auth(auth_request):
    token = auth_request.token
    if token == 'allow':
        return AuthResponse(routes=['/protected'], principal_id='user')
    return AuthResponse(routes=[], principal_id='user')


@app.route('/protected', authorizer=auth)
def index():
    return {'hello': 'world'}
```

When deployed: 
```
curl --location --request GET 'https://{{base_url}}/api/protected?queryparams=true' \
> --header 'Authorization: allow
```
results in status 200 `{"hello":"world"}`

When run in local:
```
curl --location --request GET 'http://127.0.0.1:8000/protected?queryparams=true' \
--header 'Authorization: allow'
```
results in status 403 `{"Message": "User is not authorized to access this resource"}`

Cause of the problem seems to be in :
`chalice.local.LocalARNBuilder.build_arn` method, because it would return ARN preserving the queryparameters
eg: `'arn:aws:execute-api:mars-west-1:123456789012:ymy8tbxw7b/api/GET/protected?queryparams=true'`

Alternatively if this is desired behaviour for local: 
`chalice.local.ARNMatcher._resource_match` on the line:

`resource_regex = '^%s$' % resource_regex` 

could be changed by removing `$` so it stops unmatching the remaining queryparameters that are in ARN

**Comment by jamesls** [comment]
Thanks for reporting, I'll take a look at your PR.

**Comment by jamesls** [comment]
Merged #1474
